extension/src/openvic-extension/utility/StringLiteral.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <string_view>
+#include <type_traits>
+
+namespace OpenVic {
+	template<std::size_t N>
+	struct StringLiteral {
+		constexpr StringLiteral(const char (&str)[N]) {
+			std::copy_n(str, N, value);
+		}
+
+		constexpr StringLiteral(std::string_view string) {
+			std::copy_n(string.begin(), N, value);
+		}
+
+		char value[N];
+		static const constexpr std::integral_constant<std::size_t, N - 1> size {};
+
+		struct iterator {
```

Comment:
Strictly speaking this struct is unnecessary. char* satisfies the requirements of an iterator, and so begin and end could just return the relevant pointers, given that this iterator doesn't provide any functionality beyond that of char*